### PR TITLE
Add OnDemandOrdering proxy

### PR DESCRIPTION
**Pull request by alexgparity**

### CODE REVIEW

File: runtime/kusama/src/lib.rs [hunk 1]
```diff
@@ -943,6 +943,7 @@ pub enum ProxyType {
 	Auction,
 	Society,
 	NominationPools,
+	OnDemandBidding,
```
Comment (eskimor) [inline runtime/kusama/src/lib.rs:946]: Ordering would be a better name.

File: runtime/westend/src/lib.rs [hunk 2]
```diff
@@ -768,6 +768,7 @@ pub enum ProxyType {
 	CancelProxy,
 	Auction,
 	NominationPools,
+	OnDemaindBidding,
```
Comment (eskimor) [inline runtime/westend/src/lib.rs:771]: Typo?

File: runtime/westend/src/lib.rs [hunk 3]
```diff
@@ -852,6 +853,7 @@ impl InstanceFilter<RuntimeCall> for ProxyType {
 					RuntimeCall::Registrar(..) |
 					RuntimeCall::Slots(..)
 			),
+			ProxyType::OnDemaindBidding => matches!(c, RuntimeCall::Scheduler(..)),
```
Comment (eskimor) [inline runtime/westend/src/lib.rs:856]: But consistent :sunglasses:

File: runtime/rococo/src/lib.rs [hunk 4]
```diff
@@ -976,6 +977,7 @@ impl InstanceFilter<RuntimeCall> for ProxyType {
 					RuntimeCall::Slots { .. }
 			),
 			ProxyType::Society => matches!(c, RuntimeCall::Society(..)),
+			ProxyType::OnDemandBidding => matches!(c, RuntimeCall::Scheduler(..)),
```
Comment (eskimor) [inline runtime/rococo/src/lib.rs:980]: Would be better to limit this to the actual parathread assignment provider. @antonva can we make this more restricted easily? / Does this even work, as allowing calls to the parathread assigner pallet?

Comment (alexgparity) [reply runtime/rococo/src/lib.rs]: > I think we should be matching `RuntimeCall::OnDemandProvider`

I didn't see an `OnDemandProvider` exposed at the `runtime/lib.rs` level. And still don't see one.

> not have the `ProxyType` at all in runtimes that don't implement the on demand pallet.

Which ones are these?

Comment (antonva) [reply runtime/rococo/src/lib.rs]: >I didn't see an OnDemandProvider exposed at the runtime/lib.rs level. And still don't see one.

Ah, sorry, I meant OnDemandAssignmentProvider, which should be defined here: https://github.com/paritytech/polkadot/blob/5dec26d8e199545d6c39c9429532a8dc108746d2/runtime/rococo/src/lib.rs#L1449

> Which ones are these?

All but Rococo for now. We need to change the other runtimes when we implement the OnDemandAssignmentProvider on them. Polkadot/Kusama/Westend all just set the parachains assigner as their AssignmentProvider a la: 
https://github.com/paritytech/polkadot/blob/5dec26d8e199545d6c39c9429532a8dc108746d2/runtime/polkadot/src/lib.rs#L1380
https://github.com/paritytech/polkadot/blob/5dec26d8e199545d6c39c9429532a8dc108746d2/runtime/polkadot/src/lib.rs#L1141

**Review by eskimor — APPROVED** [review]

**Pull request merged**